plugins/mac-videotoolbox/encoder.c

```diff
@@ -1071,12 +1401,36 @@ bool obs_module_load(void)
 	char *name = bzalloc(name##_len + 1);                             \
 	CFStringGetFileSystemRepresentation(name##_ref, name, name##_len);
 
-		VT_DICTSTR(kVTVideoEncoderList_CodecName, codec_name);
-		if (strcmp("H.264", codec_name) != 0) {
-			bfree(codec_name);
+		const char *codec_name = NULL;
+		get_properties2_func_t properties = NULL;
+		CMVideoCodecType codec_type = 0;
+		{
+			CFNumberRef codec_type_num = CFDictionaryGetValue(
+				encoder_dict, kVTVideoEncoderList_CodecType);
+			CFNumberGetValue(codec_type_num, kCFNumberSInt32Type,
+					 &codec_type);
+		}
+
+		switch (codec_type) {
+		case kCMVideoCodecType_H264:
+			properties = vt_properties_h264;
+			codec_name = "h264";
+			break;
+#ifdef ENABLE_HEVC
+		case kCMVideoCodecType_HEVC:
+			properties = vt_properties_hevc;
+			codec_name = "hevc";
+			break;
+#endif
+			// 422 is used as a marker for all ProRes types,
+			// since the type is stored as a profile
+		case kCMVideoCodecType_AppleProRes422:
+			properties = vt_properties_prores;
+			codec_name = "prores";
+			break;
+		default:
```

Comment:
This could be simplified a bit by just settings things directly
```suggestion
		CMVideoCodecType codec_type = 0;
		{
			CFNumberRef codec_type_num = CFDictionaryGetValue(
				encoder_dict, kVTVideoEncoderList_CodecType);
			CFNumberGetValue(codec_type_num, kCFNumberSInt32Type,
					 &codec_type);
		}

		switch (codec_type) {
		case kCMVideoCodecType_H264:
			info.properties2 = vt_properties_h264_hevc;
			info.codec = "h264";
			break;
#ifdef ENABLE_HEVC
		case kCMVideoCodecType_HEVC:
			info.properties2 = vt_properties_h264_hevc;
			info.codec ="hevc";
			break;
#endif
			// 422 is used as a marker for all ProRes types,
			// since the type is stored as a profile
		case kCMVideoCodecType_AppleProRes422:
			info.properties2 = vt_properties_prores;
			info.codec = "prores";
			break;
		default:
```